integration/singularity/store.go

```diff
@@ -431,3 +436,114 @@ func (s *SingularityStore) Describe(ctx context.Context, id blob.ID) (*blob.Desc
 	}
 	return descriptor, nil
 }
+
+func (s *SingularityStore) runCleanupWorker(ctx context.Context) {
+	// Run immediately once before starting ticker
+	go func() {
+		if err := s.cleanup(context.Background()); err != nil {
+			logger.Errorf("Local store cleanup failed: %w", err)
+		}
+	}()
+
+	ticker := time.NewTicker(s.localCleanupInterval)
+cleanupLoop:
+	for {
+		select {
+		case <-ticker.C:
+			go func() {
+				if err := s.cleanup(context.Background()); err != nil {
+					logger.Errorf("Local store cleanup failed: %w", err)
+				}
+			}()
+		case <-s.closing:
+			break cleanupLoop
+		}
+	}
+}
+
+var errCleanupAlreadyRunning = errors.New("cleanup already running")
+
+func (s *SingularityStore) cleanup(ctx context.Context) error {
+	if s.cleanupActive.Load() {
+		return errCleanupAlreadyRunning
```

Comment:
Returning error here means if a cycle is running, the next one that may get started will be skipped. In the worst case, it means it can take up to 2X the configured interval for cleanup to run.

We should document this behaviour on the options, or alternatively use a mutex instead that locks and unlocks the cycle, such that the next cycle starts immediately. i.e. average cleanup interval will converge to the configured value.

Reply:
actually, this made me realize that a mutex is not even necessary, since the worker is already inside a goroutine